pulser-core/pulser/channels/eom.py

```diff
@@ -250,16 +330,26 @@ def _lightshift(
     def _rabi_freq_per_beam(
         self, rabi_frequency: float
     ) -> dict[RydbergBeam, float]:
+        shift_factor = np.sqrt(
+            self.red_shift_coeff / self.blue_shift_coeff
+            if self.limiting_beam == RydbergBeam.RED
+            else self.blue_shift_coeff / self.red_shift_coeff
+        )
         # rabi_freq = (rabi_red * rabi_blue) / (2 * int_detuning)
-        limit_rabi_freq = self.max_limiting_amp**2 / (
-            2 * self.intermediate_detuning
+        limit_rabi_freq = (
+            shift_factor
+            * self.max_limiting_amp**2
+            / (2 * self.intermediate_detuning)
         )
         # limit_rabi_freq is the maximum effective rabi frequency value
-        # below which the rabi frequency of both beams can be matched
+        # below which the lightshift can be zero
         if rabi_frequency <= limit_rabi_freq:
-            # Both beams the same rabi_freq
-            beam_amp = np.sqrt(2 * rabi_frequency * self.intermediate_detuning)
-            return {beam: beam_amp for beam in RydbergBeam}
+            base_amp = np.sqrt(2 * rabi_frequency * self.intermediate_detuning)
+            sqrt_shift_factor = np.sqrt(shift_factor)
+            return {
+                self.limiting_beam: base_amp / sqrt_shift_factor,
+                ~self.limiting_beam: base_amp * sqrt_shift_factor,
+            }
```

Comment:
I don't think that kills the lighshift... It seems to me the expression is easier:
![image](https://github.com/pasqal-io/Pulser/assets/61453516/6b9583b0-47aa-4f91-a813-4e2dbb4b415f)

Reply:
Daaamn that was stupid

Reply:
And the lightshift is actually 0 with these values 🙃 Good job and sorry !